### ISSUE TITLE
S3 GW support ListParts for adapters

### ISSUE THREAD

**Issue by N-o-Z** [treeverse/lakeFS#7600]
tracking issue to support ListParts for the following block adapters by order of importance:

1. Azure
2. GS
3. local

AWS S3 is already covered in https://github.com/treeverse/lakeFS/issues/7599

**Comment by DnOberon** [comment]
So wait - does UploadPart not work on anything but S3 then right now? Or is it ListParts that only works on S3?
<img width="763" alt="image" src="https://github.com/treeverse/lakeFS/assets/732727/5926ade8-ee17-4e59-86ea-cc7771828739">

**Comment by N-o-Z** [comment]
@DnOberon Good catch! I'll fix this

**Comment by DnOberon** [comment]
Thanks! So you’re saying I can still do part uploads using the s3 GW even
if the underlying storage is azure?

**Comment by N-o-Z** [comment]
@DnOberon Yes, there's no change in support for UploadPart, the comment was referring to ListParts and was added to UploadPart by mistake

**Comment by DnOberon** [comment]
Awesome. Thank you!
